app/main.py

```diff
@@ -1 +1,32 @@
-# write your code here
+class Animal:
+    health = 100
+    name = ""
+    hidden = False
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.alive.append(self)
+
+    def __repr__(self) -> None:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, herbivore: Herbivore) -> None:
+
+        if not herbivore.hidden and isinstance(herbivore, Herbivore):
+            herbivore.health -= 50
+        if herbivore.health <= 0:
+            herbivore.health = 0
+            Animal.alive.remove(herbivore)
```

Comment:
```suggestion
    @staticmethod
    def bite(herbivore: Herbivore) -> None:
        if not herbivore.hidden and isinstance(herbivore, Herbivore):
            herbivore.health -= 50
        if herbivore.health <= 0:
            herbivore.health = 0
            Animal.alive.remove(herbivore)
```
Напевно краще огорнути в `@staticmethod` і видалити self, так як ми не взаємодіємо з екземпляром класу Carnivore 🤷‍♀️